### PR TITLE
Improvements to online codegen

### PR DESCRIPTION
**Pull request by cbornet**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- Change openapi-online context path to "/"
- Fix openAPI generated spec
- Some minor changes

### CODE REVIEW

**Comment by cbornet** [comment]
@jmini Indeed, that would be better. I'll check if it can be done with Springfox 🤞

File: modules/openapi-generator-online/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 FROM openjdk:8-jre-alpine
-  
+
 WORKDIR /generator
 
-COPY target/openapi-generator-online-3.0.0-SNAPSHOT.jar /generator/openapi-generator-online.jar
+COPY target/*.jar /generator/openapi-generator-online.jar
```
Comment (wing328) [inline modules/openapi-generator-online/Dockerfile]: @cbornet I tried *.jar before but I got errors and that's why I used the full filename instead.

Comment (cbornet) [reply modules/openapi-generator-online/Dockerfile]: That's weird. Works on my machine 😉 . And that's also what we do in JHipster.
Did you do `mvn clean` before packaging ?

Comment (wing328) [reply modules/openapi-generator-online/Dockerfile]: (I also tested *.jar locally before and worked fine for me)

**Comment by cbornet** [comment]
@jmini I have something working for the dynamic allowableValues with sprinfox. PR incoming.

File: modules/openapi-generator-online/Dockerfile (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator-online/Dockerfile]: Here is the error reported by the CI (travis):
```
Removing intermediate container eaef9e5f5b5b
Step 3/6 : COPY target/*.jar /generator/openapi-generator-online.jar
When using COPY with more than one source file, the destination must be a directory and end with a /
```
Ref: https://travis-ci.org/OpenAPITools/openapi-generator/builds/379603374

I'll submit a PR to fix it.

Comment (cbornet) [reply modules/openapi-generator-online/Dockerfile]: So it seems there are multiple jar in the target directory. That shouldn't be the case, isn't it ?
Can you try `COPY target/openapi-generator-online*.jar /generator/openapi-generator-online.jar` so there's no need to modify the Dockerfile for each new version ?

Comment (wing328) [reply modules/openapi-generator-online/Dockerfile]: Same result:
```
Removing intermediate container 5b368abf32be
Step 3/6 : COPY target/openapi-generator-online-*.jar /generator/openapi-generator-online.jar
When using COPY with more than one source file, the destination must be a directory and end with a /
```

Let's go with the hard-coded solution for the time being as it's not something we want to automate with the highest priority.

**Pull request merged**